src/include/sof/audio/buffer.h

```diff
@@ -197,9 +197,15 @@ static inline void buffer_writeback(struct comp_buffer *buffer, uint32_t bytes)
  */
 static inline void buffer_lock(struct comp_buffer *buffer, uint32_t *flags)
 {
-	if (!buffer->inter_core)
+	if (!buffer->inter_core) {
+		/* Ignored by buffer_unlock() below, silences "may be
+		 * used uninitialized" warning.
+		 */
+		*flags = -1;
```

Comment:
i feel like cppcheck is going to get angry at this later

Reply:
Why would it?

BTW is there a best known cppcheck configuration somewhere ready to use?

Reply:
Not sure why we are passing a uint32_t flags pointer in lock() and just a uint32_t in unlock(). Both should just pass a uint32_t. This will probably squash the warnings too.

Reply:
`lock` saves the PS (Processor Stage) whereas `unlock` restores it. So passing a pointer in the former and a value in the latter makes sense to me.

Reply:
because you are passing a signed value to an unsigned type.

Reply:
Got it, thanks. I can change this to 0xffffffff or 0xcafebabe any other value you think is more appropriate. I just tried to avoid 0 to have something standing out if needed in a debug situation.

Reply:
Also regarding cppcheck config, will add to CI once there isn't a billion bugs for it to yell at. And i think 0xffffffff is probably the best

Reply:
Sorry, @marc-hb I meant the other way around. Pass a pointer to both lock() and unlock(), and I assume this also aligns with Zephyr usage ?

Reply:
@lgirdwood passing a pointer to `unlock()` would be a major API change affecting many lines of code. This is a one-line code change to silence a (valid) warning.

Reply:
Thanks for fixing the type abuse :)

Reply:
@marc-hb - I expect that using Zephyr native APIs will affect many lines of code and this is expected. We can update this in a subsequent PR though.